libsolutil/IpfsHash.cpp

```diff
@@ -53,36 +68,142 @@ string base58Encode(bytes const& _data)
 	reverse(output.begin(), output.end());
 	return output;
 }
+
+struct Chunk
+{
+	Chunk() = default;
+	Chunk(bytes _hash, size_t _size, size_t _blockSize):
+		hash(std::move(_hash)),
+		size(_size),
+		blockSize(_blockSize)
+	{}
+
+	bytes hash = {};
+	size_t size = 0;
+	size_t blockSize = 0;
+};
+
+struct Link
```

Comment:
I don't understand - in which way is the Link data structure different from the Chunk data structure? :)

Reply:
After removing links and lengths bytes, and also functions it ended up as the same structure :)

Reply:
I removed Link completely